SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracerUserInterface.cpp

```diff
@@ -32,37 +32,46 @@ TArray<FSpatialGDKSpanId> ConvertSpanIds(const TArray<FUserSpanId>& Causes)
 }
 } // namespace
 
-FUserSpanId USpatialEventTracerUserInterface::TraceEvent(UObject* WorldContextObject, const FSpatialTraceEvent& SpatialTraceEvent)
+FUserSpanId USpatialEventTracerUserInterface::TraceEvent(UObject* WorldContextObject, const FString& EventType, const FString& EventMessage,
+														 const TMap<FString, FString>& Data)
 {
 	SpatialGDK::SpatialEventTracer* EventTracer = GetEventTracer(WorldContextObject);
 	if (EventTracer == nullptr)
 	{
 		return {};
 	}
 
-	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(SpatialTraceEvent, /* Causes */ nullptr, /* NumCauses */ 0);
+	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(TCHAR_TO_ANSI(*EventType), TCHAR_TO_ANSI(*EventMessage), /* Causes */ nullptr,
+													   /* NumCauses */ 0, [Data](SpatialGDK::FSpatialTraceEventDataBuilder& EventBuilder) {
+														   for (const auto& Pair : Data)
+														   {
+															   EventBuilder.AddKeyValue(Pair.Key, Pair.Value);
+														   }
+													   });
+
 	return SpatialGDK::SpatialEventTracer::GDKSpanIdToUserSpanId(SpanId);
 }
 
-FUserSpanId USpatialEventTracerUserInterface::TraceEventBasic(UObject* WorldContextObject, FName Type, FString Message,
-															  const TArray<FUserSpanId>& Causes)
+FUserSpanId USpatialEventTracerUserInterface::TraceEvent(UObject* WorldContextObject, const FString& EventType, const FString& EventMessage,
```

Comment:
Got a screenshot of how this looks now?

Reply:
![image](https://user-images.githubusercontent.com/56311103/115733795-1d24ed00-a381-11eb-9b3a-31b1ca7500df.png)

Reply:
Heh, sorry thought this was blueprints :)